tests/Api/Category/ListCategoryIntegration.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Akeneo\Pim\tests\Api\Category;
+
+use Akeneo\Pim\Api\CategoryApi;
+use Akeneo\Pim\Pagination\PageInterface;
+use Akeneo\Pim\Pagination\ResourceCursorInterface;
+use Akeneo\Pim\tests\Api\ApiTestCase;
+
+class ListCategoryIntegration extends ApiTestCase
+{
+    public function testListPerPage()
+    {
+        $api = $this->createClient()->getCategoryApi();
+        $uriGenerator = $this->getUriGenerator();
+        $expectedCategories = $this->getExpectedCategories();
+
+        $firstPage = $api->listPerPage(2);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertNull($firstPage->getCount());
+        $this->assertNull($firstPage->getPreviousLink());
+        $this->assertNull($firstPage->getPreviousPage());
+
+        $this->assertSame($uriGenerator->generate(CategoryApi::CATEGORIES_PATH, [], [
+            'page' => 2,
+            'limit' => 2,
+            'with_count' => false,
+        ]), $firstPage->getNextLink());
+
+        $categories = $firstPage->getItems();
+        $this->assertCount(2 ,$categories);
+        $this->assertSameContent($expectedCategories[0], $categories[0]);
+        $this->assertSameContent($expectedCategories[1], $categories[1]);
+
+        $secondPage = $firstPage->getNextPage();
+        $this->assertInstanceOf(PageInterface::class, $secondPage);
+        $this->assertSame($uriGenerator->generate(CategoryApi::CATEGORIES_PATH, [], [
+            'page' => 1,
+            'limit' => 2,
+            'with_count' => false,
+        ]), $secondPage->getPreviousLink());
+
+        $categories = $secondPage->getItems();
+        $this->assertCount(2 ,$categories);
+        $this->assertSameContent($expectedCategories[2], $categories[0]);
+        $this->assertSameContent($expectedCategories[3], $categories[1]);
+
+        $lastPage = $secondPage->getNextPage();
+        $this->assertInstanceOf(PageInterface::class, $lastPage);
+        $this->assertNull($lastPage->getNextPage());
+
+        $categories = $lastPage->getItems();
+        $this->assertCount(1 ,$categories);
+        $this->assertSameContent($expectedCategories[4], $categories[0]);
+
+        $previousPage = $lastPage->getPreviousPage();
+        $this->assertInstanceOf(PageInterface::class, $previousPage);
+
+        $categories = $previousPage->getItems();
+        $this->assertCount(2 ,$categories);
+        $this->assertSameContent($expectedCategories[2], $categories[0]);
+        $this->assertSameContent($expectedCategories[3], $categories[1]);
+    }
+
+    public function testListPerPageWithCount()
+    {
+        $api = $this->createClient()->getCategoryApi();
+
+        $firstPage = $api->listPerPage(2, true);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertSame(5, $firstPage->getCount());
+    }
+
+    public function testAll()
```

Comment:
please add a test where $queryParamers in `listPerPage()` & `all()` to check if parameters are well injected in URI